### ISSUE TITLE
Focal loss

### ISSUE THREAD

**Issue by ran337287** [HotaekHan/SSTDNet#11]
Hi, I have a question about code in loss.py.  Why do you exclude the background, and only use the object labels, when computing focal loss?

**Comment by HotaekHan** [comment]
Hi :)
Classification part include the background and regression part exclude the background as you mentioned because we don't need to find where is background.

**Comment by ran337287** [comment]
Yeah, I agree with you. But what I mean is that there is ' t = one_hot_embedding(y.data, 1+self.num_classes)
        t = t[:,1:]  # exclude background' in focal loss of loss.py. It is to compute the loss of classification, and t=t[:,1:] ignores the background.

**Comment by HotaekHan** [comment]
@ran337287 that meant not ignore background while training. because it is one-hot encoding. one hot vector for background is zero vector.

**Comment by ran337287** [comment]
Oh I got it. You used sigmoid function.

**Comment by HotaekHan** [comment]
@ran337287 yes, i used sigmoid function for probability encoding. if i use softmax function, that is also same with sigmoid. because onehot vector for background is also zero vector and loss is occuring if prediction has non-zero value.
the reason why focal loss used sigmoid function is that it allocate weight for each classifier.

**Comment by ran337287** [comment]
But I think softmax is different with sigmoid.  Zero vector can be regarded as background in sigmoid, but can't in softmax. Because each element in vector after sigmoid can be considered a separate category, but it will be false in softmax.

**Comment by HotaekHan** [comment]
@ran337287 oh, yes. you are right. in this code, softmax function cant represent background. sorry :)
